File: solutions/problem_2220_minimum_bit_flips_to_convert_number/solution.cpp
/*
 * Problem 2220: Minimum Bit Flips to Convert Number
 * ===============================================
 * Difficulty: Easy
 * Tags: Bit Manipulation
 * Pattern: Bit Manipulation
 *
 * Time Complexity:  O(n) or O(log n)
 * Space Complexity: O(1)
 */

#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int minBitFlips(int start, int goal) {
        // Bit manipulation - O(n) time, O(1) space
        int result = 0;
        for (int val : start) {
            result ^= val;
        }
        return result;
    }
};

File: solutions/problem_2220_minimum_bit_flips_to_convert_number/solution.py
"""
Problem 2220: Minimum Bit Flips to Convert Number
===============================================
Difficulty: Easy
Tags: Bit Manipulation
Pattern: Bit Manipulation

Time Complexity:  O(n) or O(log n)
Space Complexity: O(1)
"""

class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        # Bit manipulation - O(n) time, O(1) space
        result = 0
        for val in start:
            result ^= val
        return result
